crates/ra_hir_def/src/per_ns.rs

```diff
@@ -9,36 +9,37 @@ use crate::{item_scope::ItemInNs, visibility::Visibility, ModuleDefId};
 
 #[derive(Debug, Copy, Clone, PartialEq, Eq)]
 pub struct PerNs {
+    pub from_glob: bool,
```

Comment:
hmm. I fear this may need to be per-namespace as well? I'm not sure though.

Reply:
also... this may be premature, but maybe we could keep this information separately, and throw it away when the def map is done, to reduce memory usage and not have this in the public struct.

Reply:
> hmm. I fear this may need to be per-namespace as well? I'm not sure though.

I'm a bit confused. Isn't `PerNs` a `per-namespace`?

Regarding the memory usage, I can work on that.

Reply:
Yes, it stores an item per namespace (types, values, macros). What I mean is that we might need to track the `from_glob` separately for the type, value and macro.

Reply:
I just realized what you meant regarding `per-namespace` (one import type per `types`, `values`, and `macros`). I'll look into how that might affect the outcome.

Reply:
The external tracking is done, but as I asked in zulip, I'm not sure if the tracking in `ExprCollector` and `DefCollector` have to be tied together somehow. I still need to look into the "import type per namespace" change.

Reply:
Done